transformer_engine/pytorch/triton/cross_entropy.py

```diff
@@ -252,6 +252,7 @@ def element_mul_kernel(
     X_ptr += program_id * X_stride
 
     # Load the gradient output value
+    grad_output_ptr += program_id
```

Comment:
This change assumes that `grad_output_ptr` is from a tensor with outer dims that match `grad_output_ptr`. This is valid when the loss outputs are not reduced (equivalent to [`torch.nn.CrossEntropyLoss`](https://docs.pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html#crossentropyloss) with `reduction='none'`). However, we also need to handle the case where `grad_output_ptr` is a scalar (equivalent to `torch.nn.CrossEntropyLoss` with `reduction='mean'`).